Improve <Show> reference (#1172)
solidjs/solid-docs: src/routes/reference/components/show.mdx

## Keyed rendering

> When the `keyed` prop is set to `true`, changes to the `when` prop—including reference changes—trigger a re-render of the `<Show>` component’s children.

Reviewer: ```suggestion
When the `keyed` is set to `true`, any change to the `when` prop — including changes in its reference — will cause the `<Show>` component to re-render its children.
```
